Realm/RLMObjectSchema.mm

```diff
@@ -281,14 +281,21 @@ + (NSArray *)propertiesForClass:(Class)objectClass isSwift:(bool)isSwiftClass {
                     // Managed lazy vars are't allowed.
                     // NOTE: Revisit this once property behaviors are implemented in Swift.
                     NSString *lazyPropertyBaseName = [self baseNameForLazySwiftProperty:propertyName];
-                    if (lazyPropertyBaseName && ![ignoredProperties containsObject:lazyPropertyBaseName]) {
-                        @throw RLMException(@"Lazy managed property '%@' is not allowed on a Realm Swift object class. Either add the property to the ignored properties list or make it non-lazy.", lazyPropertyBaseName);
+                    if (lazyPropertyBaseName) {
```

Comment:
Since this is Objective-C++ you can pull the declaration / initialization of `lazyPropertyBaseName` into the conditional:

```objc
if (NSString *lazyPropertyBaseName = [self baseNameForLazySwiftProperty:propertyName]) {
    …
}
```

This also has the advantage of limiting the scope of the variable to the `if`'s body.